### PR TITLE
[SPARK-8828][SQL][DO_NOT_MERGE] Revert the change of SPARK-5680

### PR DESCRIPTION
**Pull request by yjshen**
JIRA: https://issues.apache.org/jira/browse/SPARK-8828

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala [hunk 1]
```diff
@@ -399,7 +399,7 @@ private[spark] object SQLConf {
     isPublic = false)
 
   val USE_SQL_AGGREGATE2 = booleanConf("spark.sql.useAggregate2",
-    defaultValue = Some(true), doc = "<TODO>")
+    defaultValue = Some(false), doc = "<TODO>")
```
Comment (yjshen) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala]: I turned off new aggregate Path here intentionally to check the revert didn't crash others which I didn't find out. Would discard this when the test success

**Comment by yjshen** [comment]
Jenkins, retest this please.

**Comment by yjshen** [comment]
Jenkins, retest this please.

**Comment by yjshen** [comment]
Unrelated test fails occasionally:
```
org.apache.spark.sql.hive.execution.HiveCompatibilitySuite.partcols1
org.apache.spark.sql.hive.execution.SortMergeCompatibilitySuite.auto_sortmerge_join_16
org.apache.spark.sql.hive.execution.SortMergeCompatibilitySuite.correlationoptimizer11
```

**Comment by yjshen** [comment]
cc @rxin @yhuai

**Comment by marmbrus** [comment]
LGTM, can you revert the flag change?

**Comment by yjshen** [comment]
@marmbrus , I've revert the flag.

**Comment by yhuai** [comment]
@yjshen Can we have a test for it? So, we can make sure that we get the correct behavior after reverting those changes.

**Comment by yjshen** [comment]
@yhuai , sure, I will add one soon.

**Comment by marmbrus** [comment]
Thanks, merging to master.

**Pull request closed without merging**